app.rb

```diff
@@ -50,4 +58,36 @@ def debug_params
     list.save!
   end
   redirect back
+end
+
+post "lists/delete" do
+  debug_params
+end  
+
+post '/lists/new' do
+  debug_params
+
+  files = Dir["./data/*.md"]
+  # @files = files.map do |file|
+  idlist= 0 
+  listids = [] 
+  puts files
+  files.each do |file|
+    listids << file[7].to_i
+  end
+
+  listids = listids.sort
+  puts listids
+  for i in 0..listids.length
+    puts idlist  
+    if (idlist == listids[i] )
+        idlist = idlist + 1
+    else
+        break
+    end    
+  end
+  list =List.new(idlist)
+  list.new_file(idlist,params['list-name'])  
+  
+  redirect back
```

Comment:
this looks way too complicated. why not just set the id for the new list as `List.load_all_lists.count`?